### PR TITLE
Using google sdk to discover the project ID if it was not provided

### PR DESCRIPTION
**Pull request by dcaba**
Hi @frodenas !

  thanks so much for developing this exporter! Potential minor contribution: 
* rather than requiring a explicit project ID, if the exporter is running in GCloud, we can use google SDK to identify the current project. 
* That makes the `google.project-id` flag no longer required (but we will error and exit if we are unable to identify it successfully). 
* We also identified an unnecessary check in the main (for another argument - you are already marking it as required in the flags package you are using)
* Creating also a go module for dependency management so, with go 1.11 or newer, dependencies are automatically fetched.
  * Vendored files also updated, as tests were no longer passing: one prometheus change required a minor amend in the handler registration

   Best!

### CODE REVIEW

File: stackdriver_exporter.go [hunk 1]
```diff
@@ -134,7 +149,7 @@ func main() {
 	}
 	prometheus.MustRegister(monitoringCollector)
 
-	http.Handle(*metricsPath, prometheus.Handler())
+	http.Handle(*metricsPath, promhttp.Handler())
```
Comment (dcaba) [inline stackdriver_exporter.go]: this is required by the new prometheus client versions, @frodenas

**Comment by SuperQ** [comment]
Please rebase this against master now that we've merged go modules support in.

**Comment by dcaba** [comment]
> Please rebase this against master now that we've merged go modules support in.

Done!

**Comment by dcaba** [comment]
Hi @SuperQ ! Is there interest on getting this merged? If yes, happy to rebase again

**Comment by SuperQ** [comment]
Yes, sorry for the moving target on master, please rebase.

**Comment by dcaba** [comment]
Despite I adapted some aspects to the new shape (logger, handlers..), I may need to revisit some aspects

**Comment by dcaba** [comment]
@SuperQ , new try / rebased and adapted to your new code. I had to rollback one flag definition you moved to the collector, as the projectID is something that, with this addition, becomes dynamic (we dont want to execute the projectID discovery logic on each collector creation / handler logic), and we want to check it during the initial program bootstrap. Feel free to share feedback or request changes. Best!

**Comment by psalaberria002** [comment]
Hei! When can we expect a release for this feature?

**Pull request merged**